linux/deploy_vm/deploy_vm.yml

```diff
@@ -19,27 +21,100 @@
         skip_reason: "Skipped"
       when: new_vm is undefined or (not new_vm | bool)
 
-    # Change vm_username to root if it is not. And add a new user after VM deployment
-    - name: "Set user account for new VM"
-      include_tasks: set_new_vm_user_account.yml
+    - name: "Test case block"
+      block:
+        # Change vm_username to root if it is not. And add a new user after VM deployment
+        - name: "Set user account for new VM"
+          include_tasks: set_new_vm_user_account.yml
 
-    - name: "Set fact of timestamp for VM hostname"
-      ansible.builtin.set_fact:
-        hostname_timestamp: "{{ lookup('pipe', 'date +%Y%m%d%H%M%S') }}"
+        - name: "Set fact of timestamp for VM hostname"
+          ansible.builtin.set_fact:
+            hostname_timestamp: "{{ timestamp }}"
 
-    - name: "Get SSH public key from localhost"
-      include_tasks: ../utils/get_local_ssh_public_key.yml
+        - name: "Get SSH public key from localhost"
+          include_tasks: ../utils/get_local_ssh_public_key.yml
 
-    - name: "Deploy VM by creating a new VM and install OS from ISO image on it"
-      include_tasks: deploy_vm_from_iso.yml
-      when: (vm_deploy_method is undefined) or (vm_deploy_method == "iso")
+        - name: "Create temp folder for generating unattend install ISO file"
```

Comment:
Is it possible not create a temp folder in cache folder since cache folder is created for temporary files?

Reply:
Without a new temp folder under cache folder, multiple runs will put ISOs, config files be mixed into same cache folder. It will be confusing.